### PR TITLE
iOS SDK v2.0: Trusted devices

### PR DESCRIPTION
**Pull request by albertpurnama**
NEED TO:
- Bump up Cotter Version

### CODE REVIEW

File: Source/Cotter.swift [hunk 1]
```diff
@@ -193,3 +224,15 @@ func transformCb(parent: UIViewController, cb: @escaping FinalAuthCallback) -> F
         cb(token, err)
     }
 }
+
+// TODO: Delete this soon
```
Comment (rayandrie) [inline Source/Cotter.swift:228]: remember delete

File: Source/Service/CotterAPIService.swift [hunk 2]
```diff
@@ -192,5 +200,308 @@ public class CotterAPIService {
...
+            return
+        }
+        
+        let pubKeyBase64 = CryptoUtil.keyToBase64(pubKey: pubKey)
+        print("current pubKey: \(pubKeyBase64)")
+        
+        
+        // get the trusted device status first
+        func innerCb(response: CotterResult<EnrolledMethods>) {
+            let method = "TRUSTED_DEVICE"
```
Comment (rayandrie) [inline Source/Service/CotterAPIService.swift:278]: Can possibly add this to CotterConstants

File: Source/Service/CotterAPIService.swift [hunk 3]
```diff
@@ -192,5 +200,308 @@ public class CotterAPIService {
...
+            cb(.failure(CotterAPIError.general(message: "This QR Code belongs to another user, and cannot be registered for this user.")))
+        }
+        
+        if newIssuer != CotterAPIService.shared.apiKeyID {
+             cb(.failure(CotterAPIError.general(message: "This QR Code belongs to another app, and cannot be registered for this app.")))
+        }
+        
+        let apiClient = self.apiClient()
+        
+        let event = "ENROLL_NEW_TRUSTED_DEVICE"
```
Comment (rayandrie) [inline Source/Service/CotterAPIService.swift:465]: can also add these to CotterConstants

File: Source/Service/CotterAPIService.swift [hunk 4]
```diff
@@ -192,5 +200,308 @@ public class CotterAPIService {
...
+        let method = "TRUSTED_DEVICE"
```
Comment (rayandrie) [inline Source/Service/CotterAPIService.swift:466]: can also add these to CotterConstants

File: Source/Crypto/KeyStore.swift [hunk 5]
```diff
@@ -1,36 +1,45 @@
 //
-//  File.swift
-//  CotterIOS
+//  KeyStore.swift
```
Comment (rayandrie) [inline Source/Crypto/KeyStore.swift:2]: Now I see your affinity for interfaces and polymorphism...

**Comment by rayandrie** [comment]
rebased version with comments changes here: https://github.com/cotterapp/ios-sdk/pull/57

**Comment by albertpurnama** [comment]
Moved to #57

**Pull request closed without merging**